toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -922,14 +942,110 @@ func createUserWithPassword(installChroot *safechroot.Chroot, user configuration
 
 	// Update password expiration
 	if user.PasswordExpiresDays != 0 {
+		// Ignore updating if there is no shadow file to update
+		if _, err2 := os.Stat(shadowFile); err2 != nil {
+			logger.Log.Debugf("No shadow file to update. Skipping.")
+			return
+		}
+
 		err = installChroot.UnsafeRun(func() error {
-			return shell.ExecuteLive(squashErrors, "chage", "-M", strconv.FormatUint(user.PasswordExpiresDays, passwordExpiresBase), user.Name)
+			return chage(strconv.FormatUint(user.PasswordExpiresDays, passwordExpiresBase), user.Name)
 		})
 	}
 
 	return
 }
 
+// chage works in the same way as invoking "chage -M passwordExpirationInDays username"
+// i.e. it sets the maximum password expiration date.
+func chage(passwordExpirationInDays string, username string) error {
+	var (
+		shadow             []string
+		passwordExpiration int64
+		err                error
+		usernameWithColon  = fmt.Sprintf("%s:", username)
+	)
+
+	shadow, err = file.ReadLines(shadowFile)
+	if err != nil {
+		return err
+	}
+
+	passwordExpiration, err = strconv.ParseInt(passwordExpirationInDays, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	for n, entry := range shadow {
+		done := false
+		// Entries in shadow are separated by colon and start with a username
+		// Finding one that starts like that means we've found our entry
+		if strings.HasPrefix(entry, usernameWithColon) {
+			// Each line in shadow contains 9 fields separated by colon ("") in the following order:
+			// login name, encrypted password, date of last password change,
+			// minimum password age, maximum password age, password warning period,
+			// password inactivity period, account expiration date, reserved field for future use
+			const (
+				loginNameField         = 0
```

Comment:
I thought about using `iota`, but since this is not an enumeration, the fields have the numbers assigned to them and they are immutable - I think it's better to be explicit about position of each field.